src/dsp_tools/utils/xml_upload_stash.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import annotations
+
+import json
+from collections import namedtuple
+from typing import Any
+
+import regex
+
+from dsp_tools.models.connection import Connection
+from dsp_tools.models.exceptions import BaseError
+from dsp_tools.models.resource import KnoraStandoffXmlEncoder
+from dsp_tools.models.value import KnoraStandoffXml
+from dsp_tools.models.xmlproperty import XMLProperty
+from dsp_tools.models.xmlresource import XMLResource
+from dsp_tools.utils.create_logger import get_logger
+from dsp_tools.utils.shared import try_network_action
+
+MetricRecord = namedtuple("MetricRecord", ["res_id", "filetype", "filesize_mb", "event", "duration_ms", "mb_per_sec"])
```

Comment:
This isn't used in the current file. And even if it was, I would rather import it from `xml_upload.py`, instead of redefining it here.